oracle/oracle.go

```diff
@@ -0,0 +1,377 @@
+	}
+
+	zones := []string{}
+	for _, placementConfig := range nodePoolDetails.NodePool.NodeConfigDetails.PlacementConfigs {
+		zones = append(zones, *placementConfig.AvailabilityDomain)
+	}
+	size := int64(*nodePoolDetails.NodeConfigDetails.Size)
+
+	return &cloudops.InstanceGroupInfo{
+		CloudResourceInfo: cloudops.CloudResourceInfo{
+			Name:   *nodePoolDetails.Name,
+			ID:     o.poolID,
+			Region: o.region,
+		},
+		Min:   &size,
+		Max:   &size,
+		Zones: zones,
+	}, nil
+}
+
+func (o *oracleOps) Describe() (interface{}, error) {
+	getInstanceReq := core.GetInstanceRequest{
+		InstanceId: &o.instance,
+	}
+	resp, err := o.compute.GetInstance(context.Background(), getInstanceReq)
+	if err != nil {
+		return nil, err
+	}
+	return resp.Instance, nil
+}
+
+func (o *oracleOps) DeviceMappings() (map[string]string, error) {
+	m := make(map[string]string)
+	var devicePath, volID string
+	volumeAttachmentReq := core.ListVolumeAttachmentsRequest{
+		InstanceId: common.String(o.instance),
+	}
+	volumeAttachmentResp, err := o.compute.ListVolumeAttachments(context.Background(), volumeAttachmentReq)
+	if err != nil {
+		return m, err
+	}
+	for _, va := range volumeAttachmentResp.Items {
+		if va.GetDevice() != nil && va.GetVolumeId() != nil {
+			devicePath = *va.GetDevice()
+			volID = *va.GetVolumeId()
+		} else {
+			logrus.Warnf("Device path or volume id for [%+v] volume attachment not found", va)
+			continue
+		}
+		m[devicePath] = volID
+	}
+	return m, nil
+}
+
+func (o *oracleOps) DevicePath(volumeID string) (string, error) {
+	volumeAttachmentReq := core.ListVolumeAttachmentsRequest{
+		VolumeId: common.String(volumeID),
+	}
+	volumeAttachmentResp, err := o.compute.ListVolumeAttachments(context.Background(), volumeAttachmentReq)
+	if err != nil {
+		return "", err
+	}
+
+	if volumeAttachmentResp.Items == nil || len(volumeAttachmentResp.Items) == 0 {
+		return "", cloudops.NewStorageError(cloudops.ErrVolDetached,
+			"Volume is detached", volumeID)
+	}
+	volumeAttachment := volumeAttachmentResp.Items[0]
+	if volumeAttachment.GetInstanceId() == nil {
+		return "", cloudops.NewStorageError(cloudops.ErrVolInval,
+			"Unable to determine volume instance attachment", "")
+	}
+	if o.instance != *volumeAttachment.GetInstanceId() {
+		return "", cloudops.NewStorageError(cloudops.ErrVolAttachedOnRemoteNode,
+			fmt.Sprintf("Volume attached on %q current instance %q",
+				*volumeAttachment.GetInstanceId(), o.instance),
+			*volumeAttachment.GetInstanceId())
+	}
+
+	if volumeAttachment.GetLifecycleState() != core.VolumeAttachmentLifecycleStateAttached {
+		return "", cloudops.NewStorageError(cloudops.ErrVolInval,
+			fmt.Sprintf("Invalid state %q, volume is not attached",
+				volumeAttachment.GetLifecycleState()), "")
+	}
+	if volumeAttachment.GetDevice() == nil {
+		return "", cloudops.NewStorageError(cloudops.ErrVolInval,
+			"Unable to determine volume attachment path", "")
+	}
+	// TODO: check do we really need to get actual device path.
```

Comment:
Can we remove this comment since we agreed on using consistent paths?

Reply:
Done